src/workerd/util/sentry.h

```diff
@@ -47,4 +48,42 @@ inline kj::StringPtr maybeOmitColoFromSentry(uint32_t coloId) {
     KJ_LOG(severity,  __VA_ARGS__); \
   }
 
-}  // namespace workerd
+// Log this to Sentry once ever per process. Typically will be better to use LOG_ERROR_PERIODICALLY.
+#define LOG_ERROR_ONCE(msg, ...)                                               \
+  do {                                                                         \
+    static bool logOnce KJ_UNUSED = [&]() {                                    \
+      KJ_LOG(ERROR, msg, ##__VA_ARGS__);                                       \
+      return true;                                                             \
+    }();                                                                       \
+  } while (0)
+
+#define LOG_ERROR_ONCE_IF(cond, msg, ...)                                      \
+  do {                                                                         \
+    if (cond) {                                                                \
```

Comment:
I wouldn't worry about it for this effort, but this pattern will make us pay for expensive conditions even if we have already logged once.

Reply:
How would you improve it?

Reply:
I suspect there will be differing opinions but my first guess was something like this:
```
#define DO_ONCE(code) ...
#define LOG_ONCE(msg, ...) DO_ONCE({ LOG(msg, ##__VA_ARGS__); })
#define LOG_ONCE_IF(cond, msg, ...) DO_ONCE({ if(cond) { LOG(msg, ##__VA_ARGS__); } })

```

Reply:
I could be mistaken, but doesn't your variant only check the condition once? That's semantically different from logging the first time the invariant is violated I think and it seems like the latter is what is being attempted in all existing cases I found.

Reply:
Ahhhhh, you are right. Yeah, this gets way trickier then. I could phrase a similar pattern to the log periodically problem by predefining the bool for "once" but it's either thread local or one of a few "safe" atomic access patterns.